src/shogun/kernel/LinearARDKernel.h

```diff
@@ -77,30 +120,34 @@ class CLinearARDKernel: public CDotKernel
 	 */
 	virtual EFeatureType get_feature_type() { return F_DREAL; }
 
-	/** set weight of particular feature/dimension
+	/** return current feature/dimension weights in matrix form
+	 * Note that a diagonal matrix weights is considered as a column matrix (p-by-one matrix)
+	 * where p is the number of features
+	 * Note that a scalar weights is considered as a one-by-one matrix
 	 *
-	 * @param w weight to set
-	 * @param i index of feature
-	 */
-	virtual void set_weight(float64_t w, index_t i);
+	 * @return	weights in matrix form */
+	virtual SGMatrix<float64_t> get_weights();
 
-	/** get weight of particular feature/dimension
-	 *
-	 * @param i index of feature
+	/** setter for feature/dimension weights in vector form
```

Comment:
Mmmmh, we should definitely have helper methods that you can give the intuitive representation that then wrap the call to this method. Could you add that in some next patch?